### ISSUE TITLE
Adding child entities : Order preserved?

### ISSUE THREAD

**Issue by xenophilios** [dotnet/EntityFramework.Docs#3855]
Please clarify whether the order of child entities is preserved when they are entered into the database. Thank you!
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0f0fb981-4dec-7ff7-817b-6606f620b875
* Version Independent ID: c7517e7d-f011-a24c-53da-913771913998
* Content: [Saving Related Data - EF Core](https://docs.microsoft.com/en-us/ef/core/saving/related-data)
* Content Source: [entity-framework/core/saving/related-data.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/saving/related-data.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @ajcvickers
* Microsoft Alias: **avickers**

**Comment by roji** [comment]
In relational databases, the order isn't preserved in dependents unless an explicit ordering column is specified. This is covered by https://github.com/dotnet/efcore/issues/9067.

**Comment by xenophilios** [comment]
Thank you for the fast response!

**Comment by roji** [comment]
Keeping open to consider documenting this, since it comes up every now and then.

**Comment by roji** [comment]
@xenophilios no. You need to have an additional int column for ordering, which you would set when adding/updating rows, and sort on in queries. That is simply how relational database work.

https://github.com/dotnet/efcore/issues/9067 is about EF helping the user with that.

**Comment by xenophilios** [comment]
Thank you for the information and the prompt response.
